# MetaSage

A machine learning–based framework designed to systematically infer regulatory mechanisms underlying metabolic dysregulation in different conditions

## Usage

3 python scripts are provided, corresponding to the 3 major steps of MetaSage:

```shell
Feature_generation.py
```

This script generates per-metabolite input files for downstream model training. For each target metabolite, the output file contains:
- The abundance of the target metabolite
- Multi-omics–derived features associated with that metabolite

The script requires the following input files:
1. gene_expression_file
Gene expression matrix derived from omics data (e.g., RNA-seq or proteomics).
   - Rows: genes (gene symbols)
   - Columns: samples (sample IDs in the first row)

2. metabolite_expression_file
Metabolite expression matrix from metabolomics data.
   - Rows: metabolites (unified metabolite names)
   - Columns: samples (sample IDs in the first row)

3. meta_gene_relation_file
A curated mapping file describing, for each target metabolite:
   - Quantified associated genes
   - Upstream reactants
   - These relationships are derived from known genome-scale metabolic models (GEMs) and filtered based on the study-specific multi-omics datasets.

4. ESTIMATE_score_results
A matrix containing four inferred tumor microenvironment scores generated by the ESTIMATE algorithm:
   - Stromal score
   - Immune score
   - ESTIMATE score
   - Tumor purity

```shell
Predictability_assessment.py
```
This script implements an XGBoost-based regression model to assess the predictability of each target metabolite.

- **Input**: feature files generated in the last step

- **Output**: a .tsv file summarizing model perfromance, including the coefficients and p-values of Peasron correlation between the observed and predicted metabolite abundance from 5-fold cross-validation.

## Citation

```
MetaSage: Machine Learning-Based Prioritization of Metabolic Regulators from Multi-Omics Data
Chenwei Wang, John M. Elizarraras, Bing Zhang
```
